products/securityscanner/api.yaml

```diff
@@ -0,0 +1,159 @@
+# Copyright 2019 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+--- !ruby/object:Api::Product
+name: SecurityScanner
+display_name: Cloud Security Scanner
+versions:
+  - !ruby/object:Api::Product::Version
+    name: beta
+    base_url: https://websecurityscanner.googleapis.com/v1beta/
+scopes:
+  - https://www.googleapis.com/auth/cloud-platform
+apis_required:
+  - !ruby/object:Api::Product::ApiReference
+    name: Web Security Scanner API
+    url: https://console.cloud.google.com/apis/library/websecurityscanner.googleapis.com/
+objects:
+  - !ruby/object:Api::Resource
+    name: ScanConfig
+    base_url: projects/{{project}}/scanConfigs
+    self_link: "{{name}}"
+    update_verb: :PATCH
+    update_mask: true
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Using Cloud Security Scanner': 'https://cloud.google.com/security-scanner/docs/scanning'
+      api: 'https://cloud.google.com/security-scanner/docs/reference/rest/v1beta/projects.scanConfigs'
+    description: |
+      A ScanConfig resource contains the configurations to launch a scan.
+    properties:
+      - !ruby/object:Api::Type::String
+        name: name
+        output: true
+        description: |
+          A server defined name for this index. Format:
+          `projects/{{project}}/scanConfigs/{{server_generated_id}}`
+      - !ruby/object:Api::Type::String
+        name: displayName
+        required: true
+        description: |
+          The user provider display name of the ScanConfig.
+      - !ruby/object:Api::Type::Integer
+        name: maxQps
+        description: |
+          The maximum QPS during scanning. A valid value ranges from 5 to 20 inclusively.
```

Comment:
In Terraform, we can use https://github.com/hashicorp/terraform/blob/master/helper/validation/validation.go#L51 to make sure that values are within a valid range.

Reply:
Do we have ways to enforce this from MM? I don't see any information on this in the overrides or property itself

Reply:
https://github.com/GoogleCloudPlatform/magic-modules/blob/master/overrides/terraform/property_override.rb#L49 which... isn't validated at all.

https://github.com/GoogleCloudPlatform/magic-modules/blob/97a4f962a6038c4fefac1da63a54237b4eced038/products/cloudscheduler/terraform.yaml#L40 is an example of using a function with it.

You can add it like this: https://github.com/terraform-providers/terraform-provider-google/blob/0d6b100f2dbed1550e486731f0a8362e6d21d81e/google/resource_storage_transfer_job.go#L201